hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFileReader.java

```diff
@@ -152,98 +153,107 @@ private void addShutDownHook() {
   // TODO : convert content and block length to long by using ByteBuffer, raw byte [] allows
   // for max of Integer size
   private HoodieLogBlock readBlock() throws IOException {
-    int blockSize;
-    long blockStartPos = inputStream.getPos();
-    try {
-      // 1 Read the total size of the block
-      blockSize = (int) inputStream.readLong();
-    } catch (EOFException | CorruptedLogFileException e) {
-      // An exception reading any of the above indicates a corrupt block
-      // Create a corrupt block by finding the next MAGIC marker or EOF
-      return createCorruptBlock(blockStartPos);
-    }
-
-    // We may have had a crash which could have written this block partially
-    // Skip blockSize in the stream and we should either find a sync marker (start of the next
-    // block) or EOF. If we did not find either of it, then this block is a corrupted block.
-    boolean isCorrupted = isBlockCorrupted(blockSize);
-    if (isCorrupted) {
-      return createCorruptBlock(blockStartPos);
-    }
-
-    // 2. Read the version for this log format
-    HoodieLogFormat.LogFormatVersion nextBlockVersion = readVersion();
+    long blockStartPos = 0;
+    long blockSize = 0;
```

Comment:
Guess line 169 ~ 172 is the line we wanna fix, can we just enclose this code snippet into the `try catch` block?
It does not seem right we also include the data block construction from line 212 ~ 246 into the `try catch`.

Reply:
yeah, I also feel we can keep line 212 to 246 outside of try catch where we deduce corruptness

Reply:
The reason the lines 212 to 246 are within the trey-catch is due to scope of variables. If I move those lines out of try-catch, I would have to define all the variables (blockType, header, footer, content) before the try so that they will be in scope after the try-catch block ends. 

This seemed an ugly code to me.

private HoodieLogBlock readBlock() throws IOException {
    long blockStartPos = 0;
    long blockSize = 0;
    HoodieLogBlockType blockType;
    HoodieLogBlock.HoodieLogBlockContentLocation logBlockContentLoc;
    Map<HeaderMetadataType, String> header;

    try {
         .....
         .....
     } catch()
          ...
     }
  
     switch(...)

Reply:
Does moving the whole try catch code into a separate method makes sense to you?

Reply:
hey Danny: for your suggestion on moving whole try catch into separate method, then the readBlock() itself may not have anything much left.so, probably we can leave it as is